core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -133,6 +122,79 @@ class ExecutorSuite extends SparkFunSuite {
       }
     }
   }
+
+  test("Gracefully handle error in task deserialization") {
+    val conf = new SparkConf
+    val serializer = new JavaSerializer(conf)
+    val env = createMockEnv(conf, serializer)
+    val serializedTask = serializer.newInstance().serialize(new NonDeserializableTask)
```

Comment:
can you take the error message as a param to NonDeserializableTask, so the magic string can be defined once and used below on line 137? From a quick read I thought the one on line 137 was coming from the Spark core code (not from the test code)